android-beacon/src/main/java/com/novoda/beacon/MessageReceiver.java

```diff
@@ -0,0 +1,58 @@
+package com.novoda.beacon;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.app.NotificationCompat;
+import android.util.Log;
+
+import com.google.android.gms.nearby.Nearby;
+import com.google.android.gms.nearby.messages.Message;
+import com.google.android.gms.nearby.messages.MessageListener;
+
+public class MessageReceiver extends BroadcastReceiver {
+
+    private static final String TAG = "!!!";
+
+    private NotificationManager notificationManager;
+    private Context context;
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+        this.context = context;
+        Nearby.Messages.handleIntent(intent, messageListener);
+    }
+
+    private final MessageListener messageListener = new MessageListener() {
+        @Override
+        public void onFound(Message message) {
+            String content = new String(message.getContent());
+            Notification notification = createNotification(context, content);
```

Comment:
Or we could trigger the service when the geofence is "close enough", i.e. within 1 km.